File: 프로그래머스/순위.cpp
#include<bits/stdc++.h>

#define SIZE 102

bool board[SIZE][SIZE];
int fight[SIZE];


using namespace std;

int solution(int n, vector<vector<int>> results) {
    int answer = 0;

    for (auto vi : results) {
        board[vi[0]][vi[1]] = true;
        fight[vi[0]]++;
        fight[vi[1]]++;
    }


    for (int k = 1; k <= n; k++) {
        for (int i = 0; i <= n; i++) {
            for (int j = 0; j <= n; j++) {
                if (board[i][j] == false && board[i][k] == true && board[k][j] == true) {
                    board[i][j] = true;
                    fight[i]++;
                    fight[j]++;
                }
            }
        }
    }

    for (int i = 1; i <= n; i++) {
        if (fight[i] == n - 1)
            answer++;
    }
    return answer;
}

File: 프로그래머스/순위.java
class Solution {
    boolean[][] graph;

    public int solution(int n, int[][] results) {
        graph = new boolean[n + 1][n + 1];

        for (int[] result : results) {
            graph[result[0]][result[1]] = true;
        }

        for (int m = 1; m <= n; m++) {
            for (int i = 1; i <= n; i++) {
                for (int j = 1; j <= n; j++) {
                    if (graph[i][m] && graph[m][j]) {
                        graph[i][j] = true;
                    }
                }
            }
        }

        int answer = 0;
        for (int i = 1; i <= n; i++) {
            int count = 0;
            for (int j = 1; j <= n; j++) {
                if (graph[i][j] || graph[j][i]) {
                    count++;
                }
            }
            if (count == n - 1) {
                answer++;
            }
        }
        return answer;
    }
}